backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/disk/image/TransferDiskImageCommand.java

```diff
@@ -825,41 +825,40 @@ private void handleCancelledSystem() {
     private void handleFinalizingSuccess(final StateContext context) {
         log.info("Finalizing successful image transfer '{}' for {}", getCommandId(), getTransferDescription());
 
-        boolean isImageVerified = true;
+        ImageStatus nextImageStatus = ImageStatus.OK;
 
         // If stopping the session did not succeed, don't change the transfer state.
         if (stopImageTransferSession(context.entity)) {
```

Comment:
I think the best solution would be not to revert but to simply fix the problem by swapping the `if` condition and also some code will be indented too the left and simplified.:
```
    private void handleFinalizingSuccess(final StateContext context) {
        log.info("Finalizing successful image transfer '{}' for {}", getCommandId(), getTransferDescription());

        // If stopping the session did not succeed, don't change the transfer state.
        if (!stopImageTransferSession(context.entity)) {
            return;
        }

        Guid transferingVdsId = context.entity.getVdsId();
        boolean isImageVerified = true;

        // Verify image is relevant only on upload.
        if (getParameters().getTransferType() == TransferType.Upload) {
            isImageVerified = verifyImage(transferingVdsId);
            if (isImageVerified) {
                // We want to use the transferring vds for image actions for having a coherent log when transferring.
                setVolumeLegalityInStorage(LEGAL_IMAGE);
                if (getDiskImage().getVolumeFormat().equals(VolumeFormat.COW)) {
                    setQcowCompat(getDiskImage().getImage(),
                            getStoragePool().getId(),
                            getDiskImage().getId(),
                            getDiskImage().getImageId(),
                            getStorageDomainId(),
                            transferingVdsId);
                    imageDao.update(getDiskImage().getImage());
                }
            }
        }

        // Finished using the image, tear it down.
        tearDownImage(context.entity.getVdsId(), context.entity.getBackupId());

        if (isImageVerified) {
            log.info("Image transfer '{}' was successful for {}", getCommandId(), getTransferDescription());
            setImageStatus(ImageStatus.OK);
            setCommandStatus(CommandStatus.SUCCEEDED);
        } else {
            setImageStatus(ImageStatus.ILLEGAL);
            updateEntityPhase(ImageTransferPhase.FINALIZING_FAILURE);
        }
    }
```

Reply:
The original code worked and was tested, when the changes were introduced originally they were cleanups and not functional changes, so we'll start by restoring the working code and then discuss cleanups (or not, as there is a long list of stuff that needs to be fixed in the negative flows in transfer image)

Reply:
OK, I think that the suggested code was more clear.
At least it might be worth to swap the `stopImageTransferSession()` condition (fail-first approach). Even such a small (and automatic by IDE) refactoring will simplify the code :)

Reply:
I prefer not to :) Once the spec is complete and we have a clearly defined behavior for each step, we can start doing refactors and cleanups because we'll know what we need to test after each change